gffquant/profilers/feature_quantifier.py

```
@@ -121,56 +154,96 @@ def process_counters(
+            self.counter,
+            self.filtered_reads,
+            gene_group_db=gene_group_db,
         )
 
-        if not external_gene_counts:
-            count_writer.write_gene_counts(
-                count_annotator.gene_counts,
-                count_annotator.scaling_factors["total_gene_uniq"],
-                count_annotator.scaling_factors["total_gene_ambi"]
+        if not gene_group_db:
+            unannotated_reads += int(round(unannotated_counts[CountMatrix.NO_ANNOTATION][0]))
```

Ruff (0.15.21)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)